### PR TITLE
obs-scripting: Replace macro adding frontend functions with plain C code

### PR DESCRIPTION
**Pull request by PatTheMav**
### Description
Replaces inline macro function to add frontend functions to the Lua script environment with a pure C-based implementation.

### Motivation and Context
Instead of using a non-debuggable macro function, keep function references and desired function names in a static array that can then be iterated over to add desired functions.

This allows for proper debugging as well as makes the combination of function and function name more explicit.

### How Has This Been Tested?
Used obs test-script used internally to check for Lua functionality (which explicitly checks for availability and return types of frontend API functions).

### Types of changes
- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: deps/obs-scripting/obs-scripting-lua-frontend.c [hunk 1]
```diff
@@ -297,37 +297,45 @@ static int add_save_callback(lua_State *script)
 
 /* ----------------------------------- */
 
+typedef struct lua_function_tuple {
+	const char *name;
+	lua_CFunction function;
+} obs_frontend_func;
+
+static obs_frontend_func functions[19] = {
```
Comment (notr1ch) [inline deps/obs-scripting/obs-scripting-lua-frontend.c:305]:
```suggestion
static const obs_frontend_func functions[] = {
```

File: deps/obs-scripting/obs-scripting-lua-frontend.c [hunk 2]
```diff
@@ -297,37 +297,45 @@ static int add_save_callback(lua_State *script)
...
-	add_func(set_current_scene_collection);
-	add_func(get_profiles);
-	add_func(get_current_profile);
-	add_func(set_current_profile);
-	add_func(remove_event_callback);
-	add_func(add_event_callback);
-	add_func(remove_save_callback);
-	add_func(add_save_callback);
-#undef add_func
+	size_t num_items = sizeof(functions) / sizeof(obs_frontend_func);
```
Comment (notr1ch) [inline deps/obs-scripting/obs-scripting-lua-frontend.c:333]: We have this macro as an alternative:
```suggestion
	size_t num_items = OBS_COUNTOF(functions);
```

File: deps/obs-scripting/obs-scripting-lua-frontend.c (hunk 1 above)
Comment (PatTheMav) [reply deps/obs-scripting/obs-scripting-lua-frontend.c]: Ah, I remember again why I did it that way - Cl.exe doesn't seem to like that:

```
D:\a\obs-studio\obs-studio\obs-studio\deps\obs-scripting\obs-scripting-lua-frontend.c(305,46): error C2099: initializer is not a constant [D:\a\obs-studio\obs-studio\obs-studio\build64\deps\obs-scripting\obs-scripting.vcxproj]
```

Comment (PatTheMav) [reply deps/obs-scripting/obs-scripting-lua-frontend.c]: It's indeed the cast - `clang` and `gcc` don't mind, but `cl` doesn't like it. Easy enough to fix.

**Pull request merged**